python-frontend/src/main/java/org/sonar/python/index/DescriptorUtils.java

```diff
@@ -115,42 +116,50 @@ public static AmbiguousDescriptor ambiguousDescriptor(AmbiguousSymbol ambiguousS
   }
 
   public static Symbol symbolFromDescriptor(Descriptor descriptor, ProjectLevelSymbolTable projectLevelSymbolTable,
```

Comment:
I think we should clean-up a bit those static methods that create symbols: this could become a Builder Class where we keep projectLevelSymbolTable, createdSymbolsByDescriptor and createdSymbolsByFqn as internal states. I don't think we need to do it in this PR, however  believe it's important to do it before releasing new version to avoid adding more technical dept. 

Let's add a TODO comment and maybe a task ticket in the Kanbar.